samcli/lib/package/artifact_exporter.py

```diff
@@ -257,65 +257,70 @@ class Resource:
+
+    def parse_property(self, original_dict, original_path, property_name, value, context):
+        if not value and not self.PACKAGE_NULL_PROPERTY:
+            return value
+        if isinstance(value, dict):
+            LOG.debug("Property %s of %s resource is not a URL", self.PROPERTY_NAME, context["resource_id"])
+            return value
         # If property is a file but not a zip file, place file in temp
         # folder and send the temp folder to be zipped
-        temp_dir = None
-        if is_local_file(property_value) and not is_zip_file(property_value) and self.FORCE_ZIP:
-            temp_dir = copy_to_temp_dir(property_value)
-            set_value_from_jmespath(resource_dict, self.PROPERTY_NAME, temp_dir)
-
+        temp_dir = False
+        if is_local_file(value) and not is_zip_file(value) and self.FORCE_ZIP:
+            temp_dir = True
+            value = copy_to_temp_dir(value)
         try:
-            self.do_export(resource_id, resource_dict, parent_dir)
-
+            return self.do_export(property_name, value, context["resource_id"], context["parent_dir"])
         except Exception as ex:
             LOG.debug("Unable to export", exc_info=ex)
             raise exceptions.ExportFailedError(
-                resource_id=resource_id, property_name=self.PROPERTY_NAME, property_value=property_value, ex=ex
+                resource_id=context["resource_id"], property_name=self.PROPERTY_NAME, property_value=value, ex=ex
             )
         finally:
             if temp_dir:
-                shutil.rmtree(temp_dir)
+                shutil.rmtree(value)
 
-    def do_export(self, resource_id, resource_dict, parent_dir):
+    def do_export(self, property_name, property_value, resource_id, parent_dir):
         """
         Default export action is to upload artifacts and set the property to
         S3 URL of the uploaded object
         If code signing configuration is provided for function/layer, uploaded artifact
         will be replaced by signed artifact location
         """
+
         # code signer only accepts files which has '.zip' extension in it
         # so package artifact with '.zip' if it is required to be signed
         should_sign_package = self.code_signer.should_sign_package(resource_id)
         artifact_extension = "zip" if should_sign_package else None
         uploaded_url = upload_local_artifacts(
-            resource_id, resource_dict, self.PROPERTY_NAME, parent_dir, self.uploader, artifact_extension
+            resource_id, property_value, self.PROPERTY_NAME, parent_dir, self.uploader, artifact_extension
         )
         if should_sign_package:
             uploaded_url = self.code_signer.sign_package(
                 resource_id, uploaded_url, self.uploader.get_version_of_artifact(uploaded_url)
             )
-        set_value_from_jmespath(resource_dict, self.PROPERTY_NAME, uploaded_url)
+
+        if self.URL_PROTOCOL == "http":
```

Comment:
How about `https`?

Reply:
This is only a way to discriminate between http and s3 (set by each resource that extends this class).